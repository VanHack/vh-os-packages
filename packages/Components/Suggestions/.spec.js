import React from 'react';
import VHSuggestions from '.';
import { render, renderer } from '../../../setupTests';

describe('[Component] - VHSuggestions', () => {
  it('renders component without crashing', () => {
    const { getByText } = render(
      <VHSuggestions title="simple title" items={[]} />
    );
    expect(getByText('simple title')).toMatchSnapshot();
  });

  it('renders VHSuggestions message when items is empty', () => {
    const json = renderer(<VHSuggestions title="simpletitle" items={[]} />).toJSON();
    const text = json.children[1].children[0];
    expect(text).toBe('Add your top skills first');
  });

  it('renders VHSuggestions items', () => {
    const {getByText} = render(
      <VHSuggestions
        title="simpletitle"
        items={[{ label: 'React.js' }, { label: 'Python' }]}
      />
    );
    expect(getByText('React.js')).toBeVisible()
    expect(getByText('Python')).toBeVisible()
  });
});
