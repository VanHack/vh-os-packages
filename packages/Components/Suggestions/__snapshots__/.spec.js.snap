// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - VHSuggestions renders component without crashing 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  color: #4f4f4f;
  font-family: Roboto;
  font-size: 14px;
  font-weight: 'regular';
  line-height: 21.5px;
}

<p
  class="c0 vh-text-component "
  color="gray-90"
>
  simple title
</p>
`;
